plugins/woocommerce/includes/data-stores/class-wc-product-data-store-cpt.php

```diff
@@ -96,6 +96,42 @@ class WC_Product_Data_Store_CPT extends WC_Data_Store_WP implements WC_Object_Da
 	 */
 	protected $updated_props = array();
 
+
+	/**
+	 * Method to obtain DB lock on SKU to make sure we only
+	 * create product with unique SKU for concurrent requests.
+	 *
+	 * We are doing so by inserting a row in the wc_product_meta_lookup table
+	 * upfront with the SKU of the product we are trying to insert.
+	 *
+	 * If the SKU is already present in the table, it means that another
+	 * request is already processing the same SKU and we should not proceed
+	 * with the insert.
+	 *
+	 * @param WC_Product $product Product object.
+	 * @return bool True if lock is obtained (unique SKU), false otherwise.
+	 */
+	private function obtain_lock_on_sku_for_concurrent_requests( $product ) {
+		global $wpdb;
+		$product_id = $product->get_id();
+		$sku        = $product->get_sku();
+
+		$query = $wpdb->prepare(
+			"INSERT INTO $wpdb->wc_product_meta_lookup (product_id, sku)
+			SELECT %d, %s
```

Comment:
Note that many hosts won't allow a select without specifying a table name, (for example when DB clustering is being used). So this needs a `FROM` statement for any existing table.

Maybe we can change this line to something like
```
SELECT %d, %s FROM $wpdb->wc_product_meta_lookup
```

Reply:
>SELECT %d, %s FROM $wpdb->wc_product_meta_lookup

used `$wpdb->options` instead of this, as tests were failing as ` $wpdb->wc_product_meta_lookup` was empty